test/index.ts

```diff
@@ -702,6 +702,43 @@ test('schema - valid set', t => {
 	});
 });
 
+test('schema - valid set with full schema', t => {
+	const schema: Schema<{foo: {bar: number; foobar: number}}> = {
+		type: 'object',
```

Comment:
Can you throw a human-friendly error if the user tries to use any other value than `object` for the `type` property? It would help non-TS users to not incorrectly use it.

Reply:
Added check and additional tests for the check